backend/src/database/querys.rs

```diff
@@ -0,0 +1,173 @@
+        .into_iter()
+        .map(|attestation| AttestationResponse::from_row(&attestation))
+        .collect()
+}
+
+pub async fn delete_attestation_request(
+    attestation_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<PgQueryResult, AppError> {
+    sqlx::query!(
+        "UPDATE attestation_requests SET deleted_at = NOW() WHERE id = $1",
+        attestation_id
+    )
+    .execute(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn insert_attestation_request(
+    attestation_request: &AttestationRequest,
+    credential: &Credential,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    let result = sqlx::query_as!(
+        AttestationResponse,
+        "WITH CreatedRow AS (INSERT INTO attestation_requests (ctype_hash, claimer, credential) VALUES ($1, $2, $3) RETURNING *) SELECT * FROM CreatedRow",
+        attestation_request.ctype_hash,
+        attestation_request.claimer,
+        serde_json::json!(credential)
+    )
+    .fetch_one(db_executor)
+    .await;
+
+    result.map_err(AppError::from)
+}
+
+pub async fn can_approve_attestation(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    sqlx::query_as!(
+        AttestationResponse,
+        "SELECT * FROM attestation_requests WHERE id = $1 AND approved = false AND revoked = false",
+        attestation_request_id
+    )
+    .fetch_one(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn approve_attestation_request(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<PgQueryResult, AppError> {
+    sqlx::query!(
+        "UPDATE attestation_requests SET approved = true WHERE id = $1",
+        attestation_request_id
+    )
+    .execute(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn can_revoke_attestation(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    sqlx::query_as!(
+        AttestationResponse,
+        "SELECT * FROM attestation_requests WHERE id = $1 AND approved = true AND revoked = false",
+        attestation_request_id
+    )
+    .fetch_one(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn revoke_attestation_request(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<PgQueryResult, AppError> {
+    sqlx::query!(
+        "UPDATE attestation_requests SET revoked = true WHERE id = $1",
+        attestation_request_id
+    )
+    .execute(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn update_attestation_request(
+    attestation_request_id: &Uuid,
+    credential: &Credential,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    sqlx::query_as!(
+        AttestationResponse,
+        "WITH UpdateRow AS (UPDATE attestation_requests SET credential = $1 WHERE id = $2 AND approved = false RETURNING *) SELECT * FROM UpdateRow",
```

Comment:
What is the difference to a simple `UPDATE attestation_requests SET credential = $1 WHERE id = $2 AND approved = false RETURNING *` ? The structure seems to be more complicated than it has to be

Reply:
Thats cool. Did not know that you can do that :fearful:  [cf41dca](https://github.com/KILTprotocol/attester/pull/1/commits/cf41dcad0e9dbf45c06ea9a3db17dd88442438d2)